### DISCUSSION TITLE
Converting Multi-size Multi-page TIFFs to PNGs

### DISCUSSION THREAD

**Discussion by usagibear** [SixLabors/ImageSharp#2379] [Q&A]
I previously asked about converting multi-page TIFFs to PNGs in this discussion (https://github.com/SixLabors/ImageSharp/discussions/2377#discussion-4916132). I have encountered issues with working with TIFFs that have multi-size pages in that one page could be say 1000px by 1000px and another could be 1234px by 5678px in the same tiff. When I encounter a TIFF like this, I get the following error: `Unhandled exception. System.NotSupportedException: Images with different sizes are not supported` upon loading the image with `Image.Load`.

There has been previous discussion and a feature on this in the following links:

https://github.com/SixLabors/ImageSharp/discussions/1982
https://github.com/SixLabors/ImageSharp/pull/2013

From these discussions, I believe I would need to do the following:

1. Decode the first frame of the Image and store it in a new image.
2. Remove the first frame and decode the second, now first frame, and store it in a different image.
3. Repeat until all the frames have been read.

However, I am not familiar with how to use the DecoderOptions and how to decode images or if the above process would work.

This is what I have now, which is broken:

```c#
DecoderOptions opt = new DecoderOptions();

using (Image image = Image.Load(opt, file))  // Breaks here
{
    for (int i = 0; i < image.Frames.Count; i++)
    {
        Console.WriteLine($"{image.Frames[i]}");

        Image page = image.Frames.CloneFrame(i);
        page.SaveAsPng($"test{i}.png");
    }
}
```

I tried using `Image.Identify()` but I can't seem to get the frame counts or frame sizes if they differ.

Any thoughts would be appreciated!

**Comment by JimBobSquarePants** [comment]
Unfortunaly that workaround is not possible. 

We simply do not support frame different sizes at this point. It's something I want to look into (I have the beginnings of a plan) but it will require a lot of work to refactor the codebase to manage this.

**Reply by usagibear** [reply]
Thanks for the quick reply! I will be following this closely. Thanks again for creating and maintaining a great project!
